src/datasource/niftizarr/frontend.ts

```diff
@@ -0,0 +1,205 @@
+import { uncancelableToken } from "#src/util/cancellation.js";
+
+
+export class NiftiZarrDataSource extends DataSourceProvider {
+  constructor(public zarrVersion: 2 | 3 | undefined = undefined) {
+    super();
+  }
+  get description() {
+    const versionStr =
+      this.zarrVersion === undefined ? "" : ` v${this.zarrVersion}`;
+    return `Nifti Zarr${versionStr} data source`;
+  }
+  get(options: GetDataSourceOptions): Promise<DataSource> {
+    // Pattern is infallible.
+    let [, providerUrl, query] =
+      options.providerUrl.match(/([^?]*)(?:\?(.*))?$/)!;
+    const parameters = parseQueryStringParameters(query || "");
+    verifyObject(parameters);
+    const dimensionSeparator = verifyOptionalObjectProperty(
+      parameters,
+      "dimension_separator",
+      parseDimensionSeparator,
+    );
+    if (providerUrl.endsWith("/")) {
+      providerUrl = providerUrl.substring(0, providerUrl.length - 1);
+    }
+    return options.chunkManager.memoize.getUncounted(
+      { type: "nifti/getVolume", providerUrl },
+      /*{
+        type: "zarr:MultiscaleVolumeChunkSource",
+        providerUrl,
+        dimensionSeparator,
+      },*/
+      async () => {
+        const { url, credentialsProvider } = parseSpecialUrl(
+          providerUrl,
+          options.credentialsManager,
+        );
+        const metadata = await getMetadata(
+          options.chunkManager,
+          credentialsProvider,
+          url,
+          {
+            zarrVersion: this.zarrVersion,
+            explicitDimensionSeparator: dimensionSeparator,
+          },
+        );
+        if (metadata === undefined) {
+          throw new Error("No zarr metadata found");
+        }
+        let multiscaleInfo: ZarrMultiscaleInfo;
+        if (metadata.nodeType === "group") {
+          // May be an OME-zarr multiscale dataset.
+          const multiscale = parseOmeMetadata(url, metadata.userAttributes);
+          if (multiscale === undefined) {
+            throw new Error("Neithre array nor OME multiscale metadata found");
+          }
+          multiscaleInfo = await resolveOmeMultiscale(
+            options.chunkManager,
+            credentialsProvider,
+            multiscale,
+            {
+              zarrVersion: metadata.zarrVersion,
+              explicitDimensionSeparator: dimensionSeparator,
+            },
+          );
+        } else {
+          multiscaleInfo = getMultiscaleInfoForSingleArray(url, metadata);
+        }
+        const volume = new MultiscaleVolumeChunkSource(
+          options.chunkManager,
+          credentialsProvider,
+          multiscaleInfo,
+        );
+        const info = await getNiftiVolumeInfo(
+          options.chunkManager,
+          credentialsProvider,
+          url,
+          uncancelableToken,
+        );
+        /*const volume = new NiftiMultiscaleVolumeChunkSource(
+          options.chunkManager,
+          credentialsProvider,
+          url,
+          info,
+        );*/
```

Comment:
Why commented-out?

Reply:
bc not sure if i did this right. there was zarr volume and nifti volume and zarr transform and nifti transform. I took volume from zarr and transform from nifti. Not sure if i was supposed to create a combo volume/transform